curator-recipes/src/main/java/org/apache/curator/framework/recipes/leader/LeaderLatch.java

```diff
@@ -667,9 +667,9 @@ protected void handleStateChange(ConnectionState newState)
             {
                 try
                 {
-                    if ( client.getConnectionStateErrorPolicy().isErrorState(ConnectionState.SUSPENDED) || !hasLeadership.get() )
```

Comment:
`getChildren` does nothing if we still hold leadership. And actually even if `hasLeadership` is true, we can say nothing if we coming back from a connection error.